api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TXCurrQueries.java

```diff
@@ -583,4 +583,23 @@ public static String getPatientWhoAfterMostRecentDateHaveDrusPickupOrConsultatio
         masterCardDrugPickup,
         artDatePickup);
   }
+
+  /**
+   * All patients marked in last “Paragen Unica (PU)” as Iniciar (I) or Continua (C) on Ficha
+   * Clinica – Master Card
+   *
+   * @return @String
+   */
+  public static String getAllPatientsMarkedInLastPuAsIOrConFichaClinicaMasterCard(
+      int encounterType,
+      int lastOnsStopConcept,
+      int startDrugsConcept,
+      int continueRegimenConcept) {
+    String query =
+        "SELECT p.patient_id FROM patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN obs o "
+            + " ON e.encounter_id=o.encounter_id WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND e.encounter_type=%d "
+            + " AND o.concept_id=%d AND o.value_coded IN(%d, %d)";
```

Comment:
Please add a filter for startDate, EndDate and Location to reduce the size of dataset being returned

Reply:
Fixed